src/main/java/appeng/client/gui/implementations/GuiInterfaceTerminal.java

```diff
@@ -79,20 +97,121 @@ public GuiInterfaceTerminal( final InventoryPlayer inventoryPlayer, final PartIn
+
+		if( btn instanceof GuiImgButton )
+		{
+			final boolean backwards = Mouse.isButtonDown( 1 );
+
+			final GuiImgButton iBtn = (GuiImgButton) btn;
+			if( iBtn.getSetting() != Settings.ACTIONS )
+			{
+				final Enum cv = iBtn.getCurrentValue();
+				final Enum next = Platform.rotateEnum( cv, backwards, iBtn.getSetting().getPossibleValues() );
+
+				if( btn == this.terminalStyleBox )
+				{
+					AEConfig.instance().getConfigManager().putSetting( iBtn.getSetting(), next );
+				}
+				else if( btn == this.searchBoxSettings )
+				{
+					AEConfig.instance().getConfigManager().putSetting( iBtn.getSetting(), next );
+				}
+				else
+				{
+					try
+					{
+						NetworkHandler.instance().sendToServer( new PacketValueConfig( iBtn.getSetting().name(), next.name() ) );
+					}
+					catch( final IOException e )
+					{
+						AELog.debug( e );
+					}
+				}
+
+				iBtn.set( next );
+
+				if( next.getClass() == SearchBoxMode.class || next.getClass() == TerminalStyle.class )
+				{
+					this.reinitalize();
+				}
+			}
+		}
+	}
+
+	private void reinitalize()
+	{
+		this.buttonList.clear();
+		this.initGui();
+		this.refreshList();
+	}
+
+	@Override
 	public void initGui()
 	{
+		final int staticSpace = 18 + 97;
+
+		calculateRows(staticSpace);
+
 		super.initGui();
 
+		this.ySize = staticSpace + this.rows * SECTION_HEIGHT;
+		final int unusedSpace = this.height - this.ySize;
+		this.guiTop = (int) Math.floor( unusedSpace / ( unusedSpace < 0 ? 3.8f : 2.0f ) );
+		int offset = this.guiTop + 8;
+
 		this.getScrollBar().setLeft( 175 );
-		this.getScrollBar().setHeight( 106 );
+		this.getScrollBar().setHeight( this.rows * SECTION_HEIGHT - 2 );
 		this.getScrollBar().setTop( 18 );
+		this.getScrollBar().setRange( 0, this.lines.size() - rows, 2 );
+
+		this.searchBoxSettings = new GuiImgButton( this.guiLeft - 18, offset, Settings.SEARCH_MODE, AEConfig.instance().getConfigManager().getSetting( Settings.SEARCH_MODE ) );
+		this.buttonList.add( this.searchBoxSettings );
+		offset += 20;
+
+		this.terminalStyleBox = new GuiImgButton( this.guiLeft - 18, offset, Settings.TERMINAL_STYLE, AEConfig.instance().getConfigManager().getSetting( Settings.TERMINAL_STYLE ) );
+		this.buttonList.add( this.terminalStyleBox );
 
 		this.searchField = new MEGuiTextField( this.fontRendererObj, this.guiLeft + Math.max( 104, this.offsetX ), this.guiTop + 4, 65, 12 );
 		this.searchField.setEnableBackgroundDrawing( false );
 		this.searchField.setMaxStringLength( 25 );
 		this.searchField.setTextColor( 0xFFFFFF );
 		this.searchField.setVisible( true );
 		this.searchField.setFocused( true );
+
+		final Enum setting = AEConfig.instance().getConfigManager().getSetting( Settings.SEARCH_MODE );
+		this.searchField.setFocused( SearchBoxMode.AUTOSEARCH == setting || SearchBoxMode.JEI_AUTOSEARCH == setting );
+
+		for( final Object s : this.inventorySlots.inventorySlots )
+		{
+			if( s instanceof AppEngSlot)
+			{
+				if( ( (Slot) s ).xDisplayPosition < 197 )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1066)